TOOLS/odoo-tools.sh

```diff
@@ -1224,23 +1227,124 @@ if [ "$SCRIPT_MODE" = "updatetranslation" ]; then
 fi
 
 # ---------------------------------------------------------------------------------------
-# $ odoo-tools.sh backup      {TARGET_BRANCH} {SUPER_PASSWORD} {DBNAME}
+# $ odoo-tools.sh backup      {TARGET_BRANCH} {DBNAME} #  [TYPE]
 # ---------------------------------------------------------------------------------------
-MODEBACKUP="odoo-tools.sh backup {TARGET_BRANCH} {SUPER_PASSWORD} {DBNAME}"
+MODEBACKUP="odoo-tools.sh backup {TARGET_BRANCH} {DBNAME|all} [odoozip|odoosql|etherpad|owncloud|full]"
 if [ "$SCRIPT_MODE" = "backup" ]; then
     echo -e "\n--------------------------------------------------------------------------------------------------------"
     echo -e " $MODEBACKUP"
     echo -e "--------------------------------------------------------------------------------------------------------"
-    if [ $# -ne 2 ]; then
-        echo -e "ERROR: \"setup-toosl.sh $SCRIPT_MODE\" takes exactly four arguments!"
+    if ! [ $# -ge 3 ] || [ $# -gt 4 ]; then
+        echo -e "ERROR: \"setup-toosl.sh $SCRIPT_MODE\" takes a minimum of three and a maximum of four arguments!"
         exit 2
     fi
 
-    # TODO: check or create INSTANCE/DATABASE/BACKUP folder
-    # Todo: CD to BACKUP folder
-    # Todo: create backup with dbname--DATE.zip with DATE in Format 2014-12-30--24-59
+    # ----- Initialize Variables
+    DBNAME=$3
+    TARGET_BRANCH=$2
+    if ! [ "x$4" = "x" ]; then
+        if [ $4 == "odoozip" ] || [ $4 == "etherpad" ] || [ $4 == "owncloud" ] || [ $4 == "full" ]; then
+            TYPE=$4
+        else
+            echo "ERROR: ${4} must be empty or in [odoozip|etherpad|owncloud|full] !"; exit 2
+        fi
+    else
+        TYPE="odoozip"
+    fi
 
+    # ----- Global Variables
+    DATETIME=`date +%Y-%m-%d__%H-%M`
+    BRANCH_PATH="${REPOPATH}/${TARGET_BRANCH}"
+    BRANCHLOGFILE="${REPOPATH}/SETUP/IS-BACKUP--${DATETIME}.log"
+    INSTANCE_PATH="${REPOPATH}/${TARGET_BRANCH}"
 
+    # ----- Find Odoo-Instance(s) to Backup
+    if [ ${DBNAME} = "all" ]; then
+        GREPREGEX="\bo8_[0-9a-zA-Z]*_[0-9a-zA-Z]*"
+    else
+        GREPREGEX="\b${DBNAME}"
+    fi
+    echo "DEBUG: regex: ${GREPREGEX}, dbname: ${DBNAME}"
+    # Todo: Use the correct linux user instead of SU except for full backup. Check rights of psql
+    declare -a DATABASES=($(su - postgres -c "psql --tuples-only -P format=unaligned -c \"SELECT datname FROM pg_database WHERE datname LIKE 'o8_%'\""|grep -o ${GREPREGEX}|sort -u))
+    if [ "x${DATABASES}" = "x" ]; then
+        echo "ERROR: No Database found!"
+    fi
+
+    # ----- HINT: Spezial case STOP Script till odoo Supports this option, service/db.py needs to be extended
+    if [ ${TYPE} = "odoosql" ]; then
```

Comment:
Please remove this block (and also remove odoosql from the comments) - since it is not working anyway.